presto-hive/src/main/java/io/prestosql/plugin/hive/HdfsConfig.java

```diff
@@ -67,6 +69,20 @@ public HdfsConfig setResourceConfigFiles(List<File> files)
         return this;
     }
 
+    @NotNull
+    @Pattern(regexp = "0[0-7]{3}", message = "must be octal number, with leading 0")
+    public String getNewDirectoryPermissions()
+    {
+        return newDirectoryPermissions;
+    }
+
+    @Config("hive.dfs.new-directory-permissions")
+    public HdfsConfig setNewDirectoryPermissions(String newDirectoryPermissions)
+    {
+        this.newDirectoryPermissions = newDirectoryPermissions;
```

Comment:
rnn?

Reply:
sure; (btw it's validated later too; there are more important properties, like the format, that are covered with validation only)